### PR TITLE
feat(SCM): support filter by name

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/scm' TESTARGS='-run=TestAccDatasourceCertificates_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/scm -v -run=TestAccDatasourceCertificates_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceCertificates_basic
=== PAUSE TestAccDatasourceCertificates_basic
=== CONT  TestAccDatasourceCertificates_basic
--- PASS: TestAccDatasourceCertificates_basic (15.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/scm       15.736s
```

### CODE REVIEW

File: docs/data-sources/scm_certificates.md [hunk 1]
```diff
@@ -43,17 +43,19 @@ The options are as follows:
...
 * `id` - The resource ID.
 
 * `certificates` - Certificate list. For details, see Data structure of the Certificate field.
-  The [Certificate](#certificate_Certificates) structure is documented below.
+  The [Certificate](#certificates_Certificate) structure is documented below.
 
-<a name="certificate_Certificates"></a>
-The `Certificates` block supports:
+<a name="certificates_Certificate"></a>
+The `Certificate` block supports:
```
Comment (Jason-Zhang9309) [inline docs/data-sources/scm_certificates.md:58]: s/Certificate/certificates

File: huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go [hunk 2]
```diff
@@ -233,17 +239,32 @@ func flattenListCertificatesBodyCertificates(resp interface{}) []interface{} {
 	return rst
 }
 
+func filterListCertificatesBodyCertificate(all []interface{}, d *schema.ResourceData) []interface{} {
+	rst := make([]interface{}, 0, len(all))
+	for _, v := range all {
+		if param, ok := d.GetOk("name"); ok && param != utils.PathSearch("name", v, nil) {
```
Comment (Lance52259) [inline huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go:245]: The judgment of `param, ok := d.GetOk("name")` can be moved outside the loop

File: huaweicloud/services/acceptance/scm/data_source_huaweicloud_scm_certificates_test.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ func TestAccDatasourceCertificates_basic(t *testing.T) {
 func testAccDatasourceCertificates_basic() string {
 	return `
 data "huaweicloud_scm_certificates" "test" {
+	name = "terrafrom-test"
 }
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/scm/data_source_huaweicloud_scm_certificates_test.go]:
```
resource "huaweicloud_scm_certificate" "test" {
  ...
}

data "huaweicloud_scm_certificates" "test" {
  depends_on = [huaweicloud_scm_certificate.test]

  name = huaweicloud_scm_certificate.test.name
}
```
Is it appropriate to not rely on a fixed certificate? (Notes: the name of certificate is not computed)

File: huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go:245]: it's too difficult to implement, and this's clear now, so keep it as it is.

File: huaweicloud/services/acceptance/scm/data_source_huaweicloud_scm_certificates_test.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/scm/data_source_huaweicloud_scm_certificates_test.go]: ok, add a preCheck.

File: huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go:245]: all right

File: docs/data-sources/scm_certificates.md [hunk 4]
```diff
@@ -43,17 +43,19 @@ The options are as follows:
 
 * `deploy_support` - (Optional, Bool) Whether to support deployment.
 
+* `name` - (Optional, String) Certificate name.
```
Comment (ShiChangkuo) [inline docs/data-sources/scm_certificates.md]: if `name` is used more commonly, suggest to put it in the front

Comment (hmbig2) [reply docs/data-sources/scm_certificates.md]: ok,done

**Pull request merged**